test/py_utility/abstract_test_case.py

```diff
@@ -28,6 +28,41 @@
 
 import inspect, os, re, shutil, sys, time, traceback, unittest
 
+class TeeFile(object):
+    """
+    A file-like object that writes to multiple destinations simultaneously.
+    Useful for capturing output while still showing it on the terminal.
+    """
+    def __init__(self, files):
+        self.files = files
+
+    def add_file(self, file):
+        self.files.append(file)
+
+    def write(self, text):
+        for f in self.files:
+            f.write(text)
+            f.flush()  # Ensure immediate output
+
+    def flush(self):
+        for f in self.files:
+            f.flush()
+
+    def fileno(self):
+        # Return the fileno of the first file
+        return self.files[0].fileno()
+
+    def close(self):
+        for f in self.files:
+            # Check if file is already closed
+            if hasattr(f, 'closed') and f.closed:
+                continue  # Skip already closed files
+            # Don't close the original stdout/stderr
+            if hasattr(f, 'fileno') and f.fileno() in (1, 2):
```

Comment:
Instead of hardcoded `(1, 2)` I recommend using `(sys.stdout.fileno(), sys.stderr.fileno())` because the standard streams can be reassigned/redirected, etc..